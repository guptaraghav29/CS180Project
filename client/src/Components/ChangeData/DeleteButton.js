import React from "react";
import Button from "@mui/material/Button";
import DeleteIcon from "@mui/icons-material/Delete";

const DeleteButton = ({ currentSelection, fetchData }) => {
  return (
    <div className="">
      <Button
        variant="contained"
        sx={{
          background: "#ff5552",
          ":hover": {
            background: "#ff3733",
          },
        }}
        startIcon={<DeleteIcon />}
        onClick={() => {
          currentSelection.forEach((id) => {
            fetch(`/cars/${id}`, { method: "DELETE" });
          });
          fetchData();
        }}
      >
        Delete
      </Button>
    </div>
  );
};

export default DeleteButton;
