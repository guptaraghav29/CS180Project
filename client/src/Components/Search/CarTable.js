import * as React from "react";
import { DataGrid } from "@mui/x-data-grid";
import DeleteButton from "../ChangeData/DeleteButton";
const columns = [
  {
    field: "id",
    headerName: "ID",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          label="tabel"
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "brand",
    headerName: "Brand",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "model",
    headerName: "Model",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "year",
    headerName: "Year",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "price",
    headerName: "Price",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "odometer",
    headerName: "Odometer",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "title_status",
    headerName: "Status",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "paint_color",
    headerName: "Color",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "region",
    headerName: "Region",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "state",
    headerName: "State",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
  {
    field: "posting_date",
    headerName: "Date Listed",
    headerAlign: "center",
    flex: 1,
    renderCell: (cellValues) => {
      return (
        <div
          style={{
            width: "100%",
            textAlign: "center",
          }}
        >
          {cellValues.value}
        </div>
      );
    },
  },
];

const CarTable = ({ carTableData, currentSelection, setCurrentSelection }) => {
  return (
    <div
      style={{
        height: 500,
        width: "100%",
        background: "#f9f9f9",
        textAlign: "center",
      }}
    >
      <DataGrid
        rows={carTableData}
        columns={columns}
        pageSize={50}
        rowsPerPageOptions={[50]}
        checkboxSelection
        sx={{
          boxShadow: 2,
          border: 0,

          "& .MuiDataGrid-row:hover": {
            background: "lightgreen",
          },
          "& .MuiDataGrid-row:nth-of-type(odd)": {
            background: "#EDEDED",
          },
          "& .MuiDataGrid-columnHeaderTitle": {
            fontWeight: "bold",
          },
          "& .MuiDataGrid-row:nth-of-type(odd):hover": {
            background: "lightgreen",
          },
        }}
        onSelectionModelChange={(ids) => {
          setCurrentSelection(ids);
        }}
      />
      <DeleteButton
        currentSelection={currentSelection}
        setCurrentSelection={setCurrentSelection}
      />
    </div>
  );
};

export default CarTable;
