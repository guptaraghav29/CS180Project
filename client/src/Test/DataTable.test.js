import * as React from "react";
import * as ReactDOM from "react-dom";
import CarTable from "../Components/Search/CarTable";
import renderer from "react-test-renderer";
import DeleteButton from "../Components/ChangeData/DeleteButton";

it("renders without any crashes", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CarTable />, div);
});

test("testing the snapshot", () => {
  const tree = renderer.create(<CarTable />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders delete button correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CarTable />, div);
  const deleteButton = div.querySelector("button");
  expect(deleteButton.innerHTML === "Delete");
});

