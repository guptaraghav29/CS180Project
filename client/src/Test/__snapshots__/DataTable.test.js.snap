// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check 1`] = `
<div
  style={
    Object {
      "background": "#f9f9f9",
      "height": 500,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    aria-colcount={12}
    aria-multiselectable={false}
    aria-rowcount={0}
    className="MuiDataGrid-root css-nj9o12-MuiDataGrid-root"
    role="grid"
  >
    <div
      className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
    >
      <div
        className="MuiDataGrid-overlay css-18rq0oa-MuiDataGrid-overlay"
        error={[TypeError: Cannot set properties of null (setting 'scrollLeft')]}
      >
        An error occurred.
      </div>
    </div>
  </div>
</div>
`;
